seqr/views/apis/staff_api_tests.py

```diff
@@ -380,3 +490,35 @@ def test_success_story(self):
 
         self.assertEqual(len(response_json['rows']), 1)
         self.assertDictEqual(response_json['rows'][0], EXPECTED_SUCCESS_STORY)
+
+    @mock.patch('seqr.views.apis.staff_api.export_multiple_files')
+    @responses.activate
+    def test_anvil_export(self, mock_export_multiple_files):
+        url = reverse(anvil_export, args=[PROJECT_GUID])
+        _check_login(self, url)
+
+        # We will test the inputs of the export_multiple_files method.
+        # Outputs of the method are not important for this test but an HttpResponse data is still required by the API.
+        mock_export_multiple_files.return_value = HttpResponse("Dummy text", content_type="text/plain")
+
+        responses.add(responses.GET, '{}/Samples'.format(AIRTABLE_URL),
+                      json=AIRTABLE_SAMPLE_RECORDS, status=200)
+        responses.add(responses.GET, '{}/Collaborator'.format(AIRTABLE_URL),
+                      json=AIRTABLE_COLLABORATOR_RECORDS, status=200)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+        mock_export_multiple_files.assert_called_with(mock.ANY,
+            u'1kg project n\u00e5me with uni\u00e7\u00f8de_AnVIL_Metadata',
+            add_header_prefix = True, file_format = 'tsv', blank_value = '-')
+
+        exported_files = mock_export_multiple_files.call_args.args[0]
+
+        self.assertListEqual([EXPECTED_PI_SUBJECT_FILE[0], EXPECTED_PI_SUBJECT_FILE[1]], [exported_files[0][0], exported_files[0][1]])
+        self.assertListEqual([EXPECTED_PI_SAMPLE_FILE[0], EXPECTED_PI_SAMPLE_FILE[1]], [exported_files[1][0], exported_files[1][1]])
+        self.assertListEqual([EXPECTED_PI_FAMILY_FILE[0], EXPECTED_PI_FAMILY_FILE[1]], [exported_files[2][0], exported_files[2][1]])
+        self.assertListEqual([EXPECTED_PI_DISCOVERY_FILE[0], EXPECTED_PI_DISCOVERY_FILE[1]], [exported_files[3][0], exported_files[3][1]])
+        self.assertIn(EXPECTED_PI_SUBJECT_FILE[2][0], exported_files[0][2])
+        self.assertIn(EXPECTED_PI_SAMPLE_FILE[2][0], exported_files[1][2])
+        self.assertIn(EXPECTED_PI_FAMILY_FILE[2][0], exported_files[2][2])
+        self.assertIn(EXPECTED_PI_DISCOVERY_FILE[2][0], exported_files[3][2])
```

Comment:
This is a bit confusing to read. Instead of having `EXPECTED_PI_<X>_FILE` be a list of 3 elements, maybe make them lists of 2 elements so the assertions can look like `self.assertListEqual(EXPECTED_PI_<X>_FILE, exported_files[X][:1])` and then make separate variables for the content so you second set of assertions would look like `self.assertIn(EXPECTED_PI_<X>_ROW, exported_files[X][2])`

Reply:
Good idea.